controllers/vspherecluster_reconciler.go

```diff
@@ -193,35 +266,82 @@ func (r *clusterReconciler) reconcileNormal(ctx context.Context, clusterCtx *cap
 
 	ok, err := r.reconcileDeploymentZones(ctx, clusterCtx)
 	if err != nil {
```

Comment:
Seems like there are subtle changes to how this conditions is set. Can you give me the reasonsing behind it so I can check if the new logic is correct?

e.g. previously the condition was set to False if noteReady > 0 (l.475)

If we did not plan to change the logic, I would prefer adding the v1beta2 conditions in the same place as v1beta1. It's much easier to ensure the logic didn't change this way (or at least for the non-error cases)

Reply:
The reasoning was to ensure this condition always gets set.
I thought that'd be better especially around when calculating the summary condition. but okay to adapt to how it was earlier.

Reply:
Changed now.

Only two changes are
* also set the condition if an error was returned.
* do not remove the condition